cmd/cmd.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package cmd
+
+import (
+	"github.com/microsoft/go-sqlcmd/cmd/root"
+	"github.com/microsoft/go-sqlcmd/internal"
+	"github.com/microsoft/go-sqlcmd/internal/cmdparser"
+	"github.com/microsoft/go-sqlcmd/internal/config"
+	"github.com/microsoft/go-sqlcmd/internal/output"
+)
+
+var loggingLevel int
+var outputType string
+var configFilename string
+var rootCmd cmdparser.Command
+
+// Initialize initializes the command-line interface. The func passed into
+// cmdparser.Initialize is called after the command-line from the user has been
+// parsed, so the helpers are initialized with the values from the command-line
+// like '-v 4' which sets the logging level to maximum etc.
+func Initialize() {
+	cmdparser.Initialize(initialize)
+	rootCmd = cmdparser.New[*Root](root.SubCommands()...)
+}
+
+func initialize() {
+	options := internal.InitializeOptions{
+		ErrorHandler: checkErr,
+		HintHandler:  displayHints,
+		OutputType:   "yaml",
+		LoggingLevel: 2,
+	}
+
+	config.SetFileName(configFilename)
```

Comment:
>configFilename [](http://example.com/codeflow?start=20&length=14)

configFilename isn't assigned any value. Is it supposed to be nul for the default with an option for the user to pass it along as a command line parameter or environment variable?

Reply:
It's set the default value for the --sqlconfig flag, so it's ~\.sqlcmd\sqlconfig (this func is called by the cobra framework after the flag defaults have been set